src/components/common/TokenAmountInput/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, Divider, FormControl, InputLabel, MenuItem, TextField } from '@mui/material'
+import { type SafeBalanceResponse } from '@safe-global/safe-gateway-typescript-sdk'
+import css from './styles.module.css'
+import NumberField from '@/components/common/NumberField'
+import { validateDecimalLength, validateLimitedAmount } from '@/utils/validation'
+import { AutocompleteItem } from '@/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer'
+import { useFormContext } from 'react-hook-form'
+import { type BigNumber } from '@ethersproject/bignumber'
+import classNames from 'classnames'
+
+export enum TokenAmountFields {
+  tokenAddress = 'tokenAddress',
+  amount = 'amount',
+}
```

Comment:
Not in this PR but I would eventually make these names parametrized.
Kudos for extracting it into a component!

Reply:
Can you give an example of what you mean with parametrized names?